### DISCUSSION TITLE
Dont`t create dynamic xfrm interface (updown: Error: Attribute failed policy validation)

### DISCUSSION THREAD

**Discussion by Viktort-rf** [strongswan/strongswan#1323] [Q&A] [answered]
**System (please complete the following information):**
 - OS: Red Hat Enterprise Linux release 8.5
 - Kernel version (if applicable): 5.4.17-2011.7.4.el8uek.x86_64
 - strongSwan version(s): U5.9.6
 - Tested/confirmed with the latest version: no

**Describe the bug**
Hello.
I'm trying to make dynamic ipsec tunnels using xfrm.
For dynamic up/down tunnel I use "updown" module and script.
Tunnels are created and work fine, but only 9 tunnels. 10th and subsequent give an error
"updown: Error: Attribute failed policy validation."
But when I stop strongswan, the logs show that it tries to manage the interfaces that should have been created
updown: Cannot find device "Virtual-Access11"
updown: Cannot find device "Virtual-Access10".
Maybe it's all about the hex xfrm policy...
Interfaces that are not created have if_id 0xa, 0xb, 0xc etc

**Expected behavior**
Creation of tunnels 10, 11 and so on, and auto updown on event SA

**Logs/Backtraces**
updown script:
```
#!/bin/bash

# set charon.install_virtual_ip = no to prevent the daemon from also installing the VIP

set -o nounset
set -o errexit

VTI_IF="Virtual-Access${PLUTO_IF_ID_OUT}"

case "${PLUTO_VERB}" in
    up-client)
        ip link add "${VTI_IF}" type xfrm dev uplink-1 \
            if_id "${PLUTO_IF_ID_OUT}"
        ip addr add 10.99.12.1/32 dev "${VTI_IF}"
        ip link set "${VTI_IF}" mtu 1360
        ip link set "${VTI_IF}" up
        ip route add "${PLUTO_PEER_SOURCEIP}" dev "${VTI_IF}"
        sysctl -w "net.ipv4.conf.${VTI_IF}.disable_policy=1"
        ;;
    down-client)
        ip link del "${VTI_IF}"
        ;;
esac
```
Swanctl.conf
```
connections {
        dmvpn {
                include /etc/strongswan/swanctl/conf.d/ike_sa_default.conf
                pools = dmvpn
                keyingtries = 1
                mobike = no
                version = 2
                remote_addrs = %any
                remote-all {
                        auth = psk
                }
                local_addrs = 203.0.113.1
                local-dmvpn {
                        auth = psk
                }
                proposals = aes256-sha512-modp2048, default
                dpd_delay = 10s
                children {
                        dmvpn {
                                include /etc/strongswan/swanctl/conf.d/child_sa_default.conf
                                updown = /etc/strongswan/swanctl/scripts/auto-interface-xfrm
                                start_action = start
                                esp_proposals = aes256-sha512, default
                                local_ts = 0.0.0.0/0
                                remote_ts = 0.0.0.0/0
                                if_id_out = %unique
                                if_id_in = %unique
                                dpd_action = clear
                        }
                }
        }
}
pools {
        dmvpn {
                addrs = 10.99.12.2-10.99.12.255
        }
}
authorities {
}
secrets {
        ike-dmvpn {
                secret = "VERY-STRONG-PSK"
        }
}
```

**Additional context**

**Comment by tobiasbrunner** [comment, marked as answer]
That interface name is too long. Linux only allows 15 characters (plus null terminator).

**Reply by Viktort-rf** [reply]
> That interface name is too long. Linux only allows 15 characters (plus null terminator).

Thank you! That helped!

**Reply by Thermi** [reply]
If you want to put any human readable name to the interface you can do it using `ip link property add dev $VTI_IF altname "some_string_without_spaces_goes_here"`

**Reply by Viktort-rf** [reply]
> If you want to put any human readable name to the interface you can do it using `ip link property add dev $VTI_IF altname "some_string_without_spaces_goes_here"`

Thanks for the information!
